skyportal/tests/api/test_photometric_series.py

```diff
@@ -0,0 +1,2176 @@
+import os
+import time
+import hashlib
+import base64
+import pytest
+import uuid
+
+import numpy as np
+import pandas as pd
+
+from astropy.time import Time
+
+import sqlalchemy as sa
+from sqlalchemy.exc import IntegrityError
+
+from skyportal.tests import api, assert_api, assert_api_fail
+from skyportal.models import DBSession, PhotometricSeries
+
+
+def convert_dataframe_to_bytes(df, metadata=None, encode=True):
+    """
+    Convert a pandas dataframe to bytes.
+
+    Parameters
+    ----------
+    df: pandas.DataFrame
+        The dataframe to convert to bytes.
+    metadata: dict
+        A dictionary of metadata to store in the HDF5 file.
+    encode: bool
+        Whether to encode the bytes as a base64 string.
+
+    Returns
+    -------
+    bytes:
+        The bytes of the dataframe.
+        If encode=True, the bytes are encoded as a base64 string
+        that can be sent over API calls.
+        If encode=False, the bytes are the raw bytes of the HDF5 file.
+    """
+
+    # this store should work without writing to disk
+    # if you open a regular store you'd just need
+    # to delete the file at the end
+    # ref: https://github.com/pandas-dev/pandas/issues/9246#issuecomment-74041497
+    with pd.HDFStore(
+        'test_file.h5', mode='w', driver="H5FD_CORE", driver_core_backing_store=0
```

Comment:
perhaps better to use a temporarynamed file here since if multiple tests are run in the same directory there could be contention over this file.